### DISCUSSION TITLE
Questions about the variable paths

### DISCUSSION THREAD

**Discussion by Zhong0501** [NVlabs/sionna#679] [Q&A]
Hi, two questions here. First, what are the objects and vertices inside the paths variable of the sionna simulation. Second, how to get all the interacting objects and interaction points of a path in sionna.

**Comment by faycalaa** [comment]
Hi @Zhong0501 ,

[Paths.vertices](https://github.com/NVlabs/sionna/blob/0aeca346148f1e47a4303f391da1db07e6df1ea7/sionna/rt/paths.py#L845) provides the coordinates of the intersection points of the rays with the scene.
[Paths.objects](https://github.com/NVlabs/sionna/blob/0aeca346148f1e47a4303f391da1db07e6df1ea7/sionna/rt/paths.py#L857) provides the [SceneObject.object_id](https://github.com/NVlabs/sionna/blob/0aeca346148f1e47a4303f391da1db07e6df1ea7/sionna/rt/scene_object.py#L71) of the intersected objects.

You can also have a look at #329.

**Comment by Zhong0501** [comment]
Thank you for your answer! I have a new question here.
Since Paths.vertices are the coordinates of the interaction points between the light and the scene, why are the coordinates of the interaction points of the reflective paths in vertices obviously out of the scope of the scene, and what's going on? I want to get the interaction point of the reflection path with the scene object, how to do it?
